src/lib/transition_router/transition_router.ml

```diff
@@ -317,12 +317,44 @@ let initialize ~logger ~network ~is_seed ~is_demo_mode ~verifier ~trust_system
       [%log info]
         "Successfully loaded frontier, but failed to download best tip from \
          network; starting participation" ;
-      return
-      @@ start_transition_frontier_controller ~logger ~trust_system ~verifier
-           ~network ~time_controller ~producer_transition_reader_ref
-           ~producer_transition_writer_ref ~verified_transition_writer
-           ~clear_reader ~collected_transitions:[] ~transition_reader_ref
-           ~transition_writer_ref ~frontier_w ~precomputed_values frontier
+      let curr_best_tip = Transition_frontier.best_tip frontier in
+      let%map () =
+        match
```

Comment:
move this to a separate function since we do the same for the catchup case below?